### PR TITLE
Fix error in sync branch due to change in GitHub API response

### PR DESCRIPTION
**Pull request by ivankristianto**
In error object, the `name` variable no longer exists.
This will cause the sync branch action to throw error:
CleanShot 2024-06-10 at 15 32 17@2x

Below is the error response object:

```
HttpError RequestError [HttpError]: Branch not found {
  status: 404,
  response: {
    url: 'https://api.github.com/repos/hm-sc/standard-chartered-assets/branches/20546-chatbot-issues-dev',
    status: 404,
    headers: {
      'access-control-allow-origin': '*',
      'access-control-expose-headers': 'ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset',
      'content-encoding': 'gzip',
      'content-security-policy': "default-src 'none'",
      'content-type': 'application/json; charset=utf-8',
      date: 'Mon, 10 Jun 2024 08:23:08 GMT',
      'referrer-policy': 'origin-when-cross-origin, strict-origin-when-cross-origin',
      server: 'GitHub.com',
      'strict-transport-security': 'max-age=31536000; includeSubdomains; preload',
      'transfer-encoding': 'chunked',
      vary: 'Accept-Encoding, Accept, X-Requested-With',
      'x-accepted-oauth-scopes': '',
      'x-content-type-options': 'nosniff',
      'x-frame-options': 'deny',
      'x-github-api-version-selected': '2022-11-28',
      'x-github-media-type': 'github.v3; format=json',
      'x-github-request-id': '15F8:C2EC0:3657BD:36DB9E:6666B7EB',
      'x-oauth-scopes': 'repo, workflow',
      'x-ratelimit-limit': '5000',
      'x-ratelimit-remaining': '4970',
      'x-ratelimit-reset': '1718010452',
      'x-ratelimit-resource': 'core',
      'x-ratelimit-used': '30',
      'x-xss-protection': '0'
    },
    data: {
      message: 'Branch not found',
      documentation_url: 'https://docs.github.com/rest/branches/branches#get-a-branch',
      status: '404'
    }
  },
  request: {
    method: 'GET',
    url: 'https://api.github.com/repos/hm-sc/standard-chartered-assets/branches/20546-chatbot-issues-dev',
    headers: {
      accept: 'application/vnd.github.v3+json',
      'user-agent': 'octokit.js/16.43.2 Node.js/20.14.0 (macOS Unknown; arm64)',
      authorization: 'token [REDACTED]'
    },
    request: { hook: [Function: bound bound register], validate: [Object] }
  }
}
```

This PR is to remove the conditional of checking the `error.name`. So we just rely on `error.status = 404`

### CODE REVIEW

File: dist/index.js [hunk 1]
```diff
@@ -2152,7 +2152,7 @@ module.exports = require("https");
...
...js --devtool source-map","build:ts":"npm run -s update-endpoints:typescript","coverage":"nyc report --reporter=html && open coverage/index.html","generate-bundle-report":"webpack-bundle-analyzer dist/bundle-stats.json --mode=static --no-open --report dist/bundle-report.html","lint":"prettier --check '{lib,plugins,scripts,test}/**/*.{js,json,ts}' 'docs/*.{js,json}' 'docs/src/**/*' index.js README.md package.json","lint:fix":"prettier --write '{lib,plugins,scripts,test}/**/*.{js,json,ts}' 'docs/*.{js,json}' 'docs/src/**/*' index.js README.md package.json","postvalidate:ts":"tsc --noEmit --target es6 test/typescript-validate.ts","prebuild:browser":"mkdirp dist/","pretest":"npm run -s lint","prevalidate:ts":"npm run -s build:ts","start-fixtures-server":"octokit-fixtures-server","test":"nyc mocha test/mocha-node-setup.js \"test/*/**/*-test.js\"","test:browser":"cypress run --browser chrome","update-endpoints":"npm-run-all update-endpoints:*","update-endpoints:fetch-json":"node scripts/update-endpoints/fetch-json","update-endpoints:typescript":"node scripts/update-endpoints/typescript","validate:ts":"tsc --target es6 --noImplicitAny index.d.ts"},"types":"index.d.ts","version":"16.43.2"};
```
Comment (stuartshields) [inline dist/index.js]: @ivankristianto looks like testing data? If we leave this in can you comment why

**Review by stuartshields — CHANGES_REQUESTED** [review]

File: dist/index.js (hunk 1 above)
Comment (ivankristianto) [reply dist/index.js]: @stuartshields no, that's the build file.

**Review by devmons — APPROVED** [review]
🚀

**Review by Rayhatron — APPROVED** [review]

**Review by stuartshields — APPROVED** [review]

**Review by svandragt — APPROVED** [review]
Looks like it will catch any error (filtered to the 404 check) now not just HTTP ones as the catch is very generic, to catch just HTTPErrors then perhaps the catch error can be made more specific. happy to merge as is and refine if we come across other errors.

**Review by stuartshields — APPROVED** [review]
Thanks for fixing that line @ivankristianto all good now

**Comment by svandragt** [comment]
status: '404', this is now a string and therefore this still raises errors.

**Pull request merged**